applications/asset_tracker_v2/src/ext_sensors/ext_sensors_bsec.c

```diff
@@ -133,81 +196,233 @@ static int enable_settings(void)
-
-	ctx.temperature_latest = (double)temperature;
-	ctx.humidity_latest = (double)humidity;
-	ctx.pressure_latest = (double)pressure / 1000;
-	ctx.air_quality_latest = (uint16_t)iaq;
-
+	for (size_t i = 0; i < n_outputs; ++i) {
+		switch (outputs[i].sensor_id) {
+		case BSEC_OUTPUT_IAQ:
+			ctx.air_quality_latest = (uint16_t) outputs[i].signal;
+			LOG_INF("IAQ: %d", ctx.air_quality_latest);
+			break;
+		case BSEC_OUTPUT_SENSOR_HEAT_COMPENSATED_TEMPERATURE:
+			ctx.temperature_latest = (double) outputs[i].signal;
+			LOG_INF("Temp: %.2f", ctx.temperature_latest);
+			break;
+		case BSEC_OUTPUT_RAW_PRESSURE:
+			ctx.pressure_latest = (double) outputs[i].signal / 1000.f;
+			LOG_INF("Press: %.2f", ctx.pressure_latest);
+			break;
+		case BSEC_OUTPUT_SENSOR_HEAT_COMPENSATED_HUMIDITY:
+			ctx.humidity_latest = (double) outputs[i].signal;
+			LOG_INF("Hum: %.2f", ctx.humidity_latest);
+			break;
+		default:
+			LOG_WRN("unknown bsec output id: %d", outputs[i].sensor_id);
+			break;
+		}
+	}
 	k_spin_unlock(&(ctx.sensor_read_lock), key);
 }
 
-static uint32_t state_load(uint8_t *state_buffer, uint32_t n_buffer)
+static size_t sensor_data_to_bsec_inputs(bsec_bme_settings_t sensor_settings,
+					 const struct bme68x_data *data,
+					 bsec_input_t *inputs, uint64_t timestamp_ns)
 {
-	if ((ctx.state_buffer_len > 0) && (ctx.state_buffer_len <= n_buffer)) {
-		memcpy(state_buffer, ctx.state_buffer, ctx.state_buffer_len);
-		return ctx.state_buffer_len;
+	size_t i = 0;
+
+	if (BSEC_INPUT_PRESENT(sensor_settings, BSEC_INPUT_TEMPERATURE)) {
+		inputs[i].sensor_id = BSEC_INPUT_HEATSOURCE;
+		inputs[i].signal = temp_offset;
+		inputs[i].time_stamp = timestamp_ns;
+		i++;
+		inputs[i].sensor_id = BSEC_INPUT_TEMPERATURE;
+		inputs[i].signal = data->temperature;
+	#ifndef BME68X_USE_FPU
+		inputs[i].signal /= 100.0f;
+	#endif
+		inputs[i].time_stamp = timestamp_ns;
+		i++;
 	}
-
-	LOG_DBG("No stored state to load");
-	return 0;
+	if (BSEC_INPUT_PRESENT(sensor_settings, BSEC_INPUT_HUMIDITY)) {
+		inputs[i].sensor_id = BSEC_INPUT_HUMIDITY;
+		inputs[i].signal =  data->humidity;
+	#ifndef BME68X_USE_FPU
+		inputs[i].signal /= 1000.0f;
+	#endif
+		inputs[i].time_stamp = timestamp_ns;
+		i++;
+	}
+	if (BSEC_INPUT_PRESENT(sensor_settings, BSEC_INPUT_PRESSURE)) {
+		inputs[i].sensor_id = BSEC_INPUT_PRESSURE;
+		inputs[i].signal =  data->pressure;
+		inputs[i].time_stamp = timestamp_ns;
+		i++;
+	}
+	if (BSEC_INPUT_PRESENT(sensor_settings, BSEC_INPUT_GASRESISTOR)) {
+		inputs[i].sensor_id = BSEC_INPUT_GASRESISTOR;
+		inputs[i].signal =  data->gas_resistance;
+		inputs[i].time_stamp = timestamp_ns;
+		i++;
+	}
+	if (BSEC_INPUT_PRESENT(sensor_settings, BSEC_INPUT_PROFILE_PART)) {
+		inputs[i].sensor_id = BSEC_INPUT_PROFILE_PART;
+		if (sensor_settings.op_mode == BME68X_FORCED_MODE) {
+			inputs[i].signal =  0;
+		} else {
+			inputs[i].signal =  data->gas_index;
+		}
+		inputs[i].time_stamp = timestamp_ns;
+		i++;
```

Comment:
This function is hard to read when it uses `i++`. Could you either comment what value is expected where or use indeces directly?

Reply:
These indices are dynamic. I cannot use them in a more direct way.